detox/src/devices/drivers/android/genycloud/services/GenyInstanceLookupService.js

```diff
@@ -1,40 +1,39 @@
+const _ = require('lodash');
 const Instance = require('./dto/GenyInstance');
 
 class GenyInstanceLookupService {
-  constructor(genyCloudExec, instanceNaming, deviceRegistry) {
+  constructor(genyCloudExec, instanceNaming, genycloudDeviceRegistry) {
     this.genyCloudExec = genyCloudExec;
     this.instanceNaming = instanceNaming;
-    this.deviceRegistry = deviceRegistry;
+    this.deviceRegistry = genycloudDeviceRegistry;
   }
 
-  async findFreeInstance(recipeUUID) {
-    const freeInstances = await this._getRelevantInstances(recipeUUID);
+  async findFreeInstance() {
+    const freeInstances = await this._getRelevantInstances();
     return (freeInstances[0] || null);
   }
 
-  async _getRelevantInstances(recipeUUID) {
+  async getInstance(instanceUUID) {
+    const instances = await this._getAllInstances();
+    return _.find(instances, (instance) => instance.uuid === instanceUUID);
+  }
+
+  async _getRelevantInstances() {
+    const takenDevices = this.deviceRegistry.getRegisteredInstanceUUIDs();
```

Comment:
More efficient to read the content once, and reuse (instead of calling `isBusy` repeatedly).